File: app/frontend/src/app/features/expedientes/models/expediente.model.ts
export interface Expediente {
  IdExpediente: number;
  CodigoExpediente?: string;
  PrimerDueno?: string;
  Ano?: number;
  FechaInicio?: string;
  FechaFin?: string;
  Estado?: string;
  Dependencia?: string;
  Caratula?: string;
  Descripcion?: string;
  Observaciones?: string;
  IdPropiedadMinera?: number;
  IdTipoExpediente: number;
  IdTransaccion?: number;
  alertas?: Alerta[];
  actas?: Acta[];
  observaciones?: import('../../observaciones/models/observacion.model').Observacion[];
}

export interface ExpedienteCreate {
  CodigoExpediente?: string;
  PrimerDueno?: string;
  Ano?: number;
  FechaInicio?: string;
  FechaFin?: string;
  Estado?: string;
  Dependencia?: string;
  Caratula?: string;
  Descripcion?: string;
  Observaciones?: string;
  IdPropiedadMinera?: number;
  IdTipoExpediente: number;
  IdTransaccion?: number;
}

export interface ExpedienteFilter {
  CodigoExpediente?: string;
  IdPropiedadMinera?: number;
}

export interface Alerta {
  idAlerta: number;
  IdTransaccion: number;
  Estado: string;
  // Agregar más campos según el modelo de alerta del backend
}

export interface Acta {
  IdActa: number;
  IdTransaccion?: number;
  IdExpediente: number;
  IdTipoActa?: string;
  Fecha?: string;
  Lugar?: string;
  IdAutoridad?: number;
  Descripcion?: string;
  Obs?: string;
  AudFecha?: string;
  AudUsuario?: number;
}

export interface ExpedienteResponse {
  data: Expediente[];
  total: number;
  range: string;
}

File: app/backend/models/expediente_model.py
from sqlalchemy import Column, ForeignKey, Integer, String, DateTime, Date, Text
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import relationship
from backend.models.tipo_expediente_model import TipoExpediente
from backend.database.connection import Base



class Expediente(Base):
    __tablename__ = "Expediente"

    IdExpediente = Column(Integer, primary_key=True, index=True, nullable=False)
    CodigoExpediente = Column(String(50), nullable=True)
    PrimerDueno = Column(String(50), nullable=True)
    Ano = Column(Integer, nullable=True)
    FechaInicio = Column(Date, nullable=True)
    FechaFin = Column(Date, nullable=True)
    Estado = Column(String(50), nullable=True)
    Dependencia = Column(String(100), nullable=True)
    Caratula = Column(String(200), nullable=True)
    Descripcion = Column(String(50), nullable=True)
    Observaciones = Column(Text, nullable=True)
    IdPropiedadMinera = Column(Integer, nullable=True)
    IdTipoExpediente = Column(Integer, ForeignKey("TipoExpediente.IdTipoExpediente"), nullable=False)
    IdTransaccion = Column(Integer, nullable=True)

    tipo_expediente = relationship("TipoExpediente", backref="expedientes")